hsuforum_actions_lib.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+// required for LIKE functionality move to hsuforum.
+
+
+/**
+* Forum Actions
+*
+* @package   hsuforum_actions
+* @copyright 2014 Moodle Pty Ltd (http://moodle.com)
+* @author    Mikhail Janowski <mikhail@getsmarter.co.za>
+* @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+*/
+
+// Add javascript to the advanced forum pages
+// if (isset($_SERVER['REQUEST_URI']) && strpos($_SERVER['REQUEST_URI'], 'mod/hsuforum/discuss.php')) {
+//    global $PAGE;
+//    // $PAGE->requires->js_call_amd('local_hsuforum_actions/hsuforum_actions', 'init');
+// }
+
+// This function is included to force moodle to run the local lib.php on every page.
+function local_hsuforum_actions_extend_navigation($param) {}
+
+function hsuforum_populate_post_actions(&$posts)
```

Comment:
Is this function, and the ones below this one still needed (not sure how/where they are called)? Would be ideal if we can include this logic where the card is rendered.

Reply:
will remove and see if it causes any issues on local

Reply:
removed function local_hsuforum_actions_extend_navigation($param) {} but the hsu_forum_populate_post_actions is used elsewhere. included in the hsuforum lib.php

Reply:
How are actions rendered on a post? Meaning does JS inject it or is an action rendered without JS in a post?